test/unit-tests/components/views/settings/encryption/ChangeRecoveryKey-test.tsx

```diff
@@ -120,27 +122,66 @@ describe("<ChangeRecoveryKey />", () => {
+            await waitFor(() => user.click(getByRole("button", { name: "Continue" })));
             // Display the form to confirm the recovery key
-            await waitFor(() => user.click(screen.getByRole("button", { name: "Continue" })));
+            await waitFor(() => user.click(getByRole("button", { name: "Continue" })));
 
-            await waitFor(() => expect(screen.getByText("Enter your recovery key to confirm")).toBeInTheDocument());
+            await waitFor(() => expect(getByText("Enter your recovery key to confirm")).toBeInTheDocument());
 
-            const finishButton = screen.getByRole("button", { name: "Finish set up" });
-            const input = screen.getByTitle("Enter recovery key");
+            const finishButton = getByRole("button", { name: "Finish set up" });
+            const input = getByTitle("Enter recovery key");
             await userEvent.type(input, "encoded private key");
+
+            await waitFor(() => {
+               expect(finishButton).not.toBeDisabled();
+            });
+
+            jest.spyOn(Modal, "createDialog");
+
             await user.click(finishButton);
 
-            await screen.findByText("Failed to set up secret storage");
-            await screen.findByText("Error: can't bootstrap");
+            expect(Modal.createDialog).toHaveBeenCalledWith(ErrorDialog, {
+                title: "Failed to set up secret storage",
+                description: "Error: can't bootstrap",
+            });
+            await waitFor(() => user.click(getByRole("button", { name: "OK" })));
+
             expect(consoleErrorSpy).toHaveBeenCalledWith(
                 "Failed to set up secret storage:",
                 new Error("can't bootstrap"),
             );
         });
+
+        it("should disallow repeated attempts to change the recovery key", async () => {
+            const mockFn = mocked(matrixClient.getCrypto()!).bootstrapSecretStorage.mockImplementation(() => {
+                // Pretend to do some work.
+                return new Promise((r) => setTimeout(r, 200));
+            });
+
+            const user = userEvent.setup();
+            const { getByRole, getByText, getByTitle } = renderComponent(false);
+
+            // Display the recovery key to save
+            await waitFor(() => user.click(getByRole("button", { name: "Continue" })));
+            // Display the form to confirm the recovery key
+            await waitFor(() => user.click(getByRole("button", { name: "Continue" })));
+
+            await waitFor(() => expect(getByText("Enter your recovery key to confirm")).toBeInTheDocument());
+
+            const finishButton = getByRole("button", { name: "Finish set up" });
+            const input = getByTitle("Enter recovery key");
+            await userEvent.type(input, "encoded private key");
+
+            await waitFor(() => {
+               expect(finishButton).not.toBeDisabled();
+            });
+
+            await user.click(finishButton);
```

Comment:
Should we do this multiple times? (How are we checking it disallows repeated attempts?)